pricegraph/src/graph/shortest_paths.rs

```diff
@@ -11,25 +11,53 @@ use petgraph::visit::{
 /// A vector associating to each node index a distance from the source.
 type Distance<G> = Vec<<G as Data>::EdgeWeight>;
 
+type PredecessorVec<G> = Vec<Option<<G as GraphBase>::NodeId>>;
+
+struct UpdatableDistance<G: Data> {
+    current: Distance<G>,
+    update: Distance<G>,
+}
+
 /// Stores the information needed to manage the predecessor list.
-/// For each node index, this type contains its predecessor node and
-/// distance in the graph from a source node.
-struct PredecessorStore<G: GraphBase + Data> {
-    predecessors: Vec<Option<G::NodeId>>,
-    distance: Distance<G>,
+/// For each node index, this type contains the information needed to
+/// compute predecessor nodes and the distance in the graph from a source
+/// node.
+enum PredecessorStore<G: GraphBase + Data> {
```

Comment:
This file could use a textual explanation in a comment about how the two approaches differ and what the different fields in this enum mean.

Reply:
I think if we split this into two separate implementations, we should include the documentation there. Also, the modified "bounded Bellman-Ford" algorithm should have some module level documentation since it is sort of invented.